### PR TITLE
scx_bpfland: improve cpufreq awareness

### PR DESCRIPTION
**Pull request by arighi**
Provide a basic cpufreq control, based on the selected EPP and scheduler's performance profile.

Improve turbo boost CPU selection logic.

### CODE REVIEW

File: scheds/rust/scx_bpfland/src/main.rs [hunk 1]
```diff
@@ -356,15 +348,27 @@ impl<'a> Scheduler<'a> {
...
-        let res =
-            File::open("/sys/devices/system/cpu/cpufreq/policy0/energy_performance_preference")
-                .and_then(|mut file| {
-                    let mut contents = String::new();
-                    file.read_to_string(&mut contents)?;
-                    Ok(contents.trim().to_string())
-                });
-
-        res.unwrap_or_else(|_| "none".to_string())
+        let energy_pref_path =
```
Comment (hodgesds) [inline scheds/rust/scx_bpfland/src/main.rs:351]: At some point maybe we can pull some of the cpu frequency governor into the topology helpers.

Comment (arighi) [reply scheds/rust/scx_bpfland/src/main.rs:351]: Yes, that's a good idea. I'll send a separate PR for that.

BTW, unrelated but similar topic, it'd be nice to include also `amd_pstate_highest_perf` to the turbo cores logic.

**Comment by htejun** [comment]
There's no "auto" mode when SCX scheduler is loaded and schedutil governor is used (there's no signal that the kernel side can use or even know). If you don't change anything, it'll just remain at the last setting.

**Comment by arighi** [comment]
Ah.. I was missing that, then I always need to set a value, I can't just ignore it (unless I actually want to keep the previous value). Ok, thanks @htejun I'll change the logic to take that aspect into account.

**Review by htejun — APPROVED** [review]

**Pull request merged**